<!DOCTYPE html>
<html lang="en"><head>
<script src="Presentation.part1and3.Heather_files/libs/clipboard/clipboard.min.js"></script>
<script src="Presentation.part1and3.Heather_files/libs/quarto-html/tabby.min.js"></script>
<script src="Presentation.part1and3.Heather_files/libs/quarto-html/popper.min.js"></script>
<script src="Presentation.part1and3.Heather_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="Presentation.part1and3.Heather_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Presentation.part1and3.Heather_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="Presentation.part1and3.Heather_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.6.42">

  <meta name="author" content="Dr.&nbsp;Heather Kates and Kalyanee Shirlekar">
  <title>Best Practices for Interpreting Omics Data with Pathway Enrichment Analysis (Part 1 and 3 edited)</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="Presentation.part1and3.Heather_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="Presentation.part1and3.Heather_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
  </style>
  <link rel="stylesheet" href="Presentation.part1and3.Heather_files/libs/revealjs/dist/theme/quarto-2f366650f320edcfcf53d73c80250a32.css">
  <link href="Presentation.part1and3.Heather_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="Presentation.part1and3.Heather_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="Presentation.part1and3.Heather_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="Presentation.part1and3.Heather_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Best Practices for Interpreting Omics Data with Pathway Enrichment Analysis (Part 1 and 3 edited)</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Dr.&nbsp;Heather Kates and Kalyanee Shirlekar 
</div>
</div>
</div>

</section>
<section class="slide level2">

<h3 id="why-this-topic-matters">Why This Topic Matters</h3>
<blockquote>
<p><strong>Pathway enrichment is powerful ‚Äî and deceptively simple.</strong></p>
</blockquote>
<hr>
<p><strong>‚úì A go-to method</strong><br>
For interpreting high-dimensional omics data</p>
<p><strong>‚úì Widely used in cancer research</strong><br>
To extract biological meaning from gene lists</p>
<hr>
<p>‚ùó <strong>But beware:</strong><br>
- Easy to misuse and misreport with unclear inputs and online tools<br>
- Overwhelming: many options for databases, methods, and cutoffs</p>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="outline">Outline</h3>
<div style="font-size: 1.1em;">
<p><strong>1. What is pathway enrichment analysis (PEA)?</strong><br>
</p>
<p><strong>2. Overview of methods: ORA, GSEA, Topology</strong><br>
</p>
<p><strong>3. Input and background sets: what they are and why they matter</strong><br>
</p>
<p><strong>4. Common pitfalls &amp; tips for better PEA</strong><br>
</p>
<p><strong>5. Visualization, reproducibility, and practical tools</strong><br>
</p>
<p><strong>6. Open Questions</strong></p>
</div>
<p><strong>Main Sources:</strong></p>
<div style="font-size: 0.6em;">
<p><a href="https://www.sciencedirect.com/science/article/pii/S0168952523000185#bb0150">Interpreting omics data with pathway enrichment analysis</a></p>
<p><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010348">Nine quick tips for pathway enrichment analysis</a></p>
</div>
<aside class="notes">
<p>Heather: Briefly walk through each step and explain how the talk will move from fundamentals, pitfalls, best practices, tools, and broader applications.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="what-is-pathway-enrichment-analysis">What Is Pathway Enrichment Analysis?</h3>
<div style="font-size: 1.1em;">
<ul>
<li><p><strong>Statistical approach</strong> to identify biological pathways whose genes show non-random patterns in an omics dataset</p></li>
<li><p><strong>Inputs:</strong> either a gene list or a full ranked list of features from an omics dataset</p></li>
<li><p><strong>Outputs:</strong> enriched biological processes or pathways</p></li>
<li><p>Common in web tools: gProfiler, Enrichr, DAVID, Reactome, ExpressAnalyst etc.</p></li>
</ul>
</div>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="three-main-methods-of-pathway-enrichment-analaysis">Three Main Methods of Pathway Enrichment Analaysis</h3>
<ul>
<li><strong>Over-representation analysis (ORA):</strong>
<ul>
<li>Uses DEGs and a background</li>
<li>Compares observed vs.&nbsp;expected overlap</li>
</ul></li>
<li><strong>Gene Set Enrichment Analysis (GSEA):</strong>
<ul>
<li>Uses a full ranked list of genes</li>
<li>Captures subtle, coordinated effects</li>
</ul></li>
<li><strong>Topology-based analysis:</strong>
<ul>
<li>Uses gene‚Äìgene relationships in pathway networks</li>
</ul></li>
</ul>
<aside class="notes">
<p>Heather: the many permutations of PEA can be grouped into three categories that each ask a different question and use a different statistical approach to answer that question.</p>
<p>This context helps to simplify the myrid online tools available for pathway enrichment analysis - the analysis that you can do with shinyGO is not fundamentally different than the analysis you can do with expressAnalyst or Enrichr, and it is critical that those tools are not methods.</p>
<p>Each tool implements one or more methods, queries one or more databases, and enables additional parameters. It‚Äôs the state of all of those options that define a pathway enrichment analysis, you can‚Äôt know exactly what an enrichment analysis means just by knowing what web tool was used to produce it.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="the-web-tool-black-box-problem">The Web Tool Black Box Problem</h3>
<p><strong>100 random unrelated genes</strong></p>
<textarea rows="10" style="width:100%">ABCA3  
ACVR1  
AHR  
AKT3  
ALG8  
AMOTL2  
ANGPTL3  
ANK2  
AOX1  
ARHGAP21  
ASB10  
ATP5MC3  
AVPR1A  
B3GNT5  
BAK1  
BCL10  
BDNF  
BLM  
BRD8  
BST2  
C1GALT1  
C3AR1  
CACNA1B  
CAMK2D  
CASR  
CCL25  
CD160  
CDC14A  
CDK10  
CELSR1  
CHD6  
CHMP5  
CLDN14  
CLRN3  
CNOT1  
COLEC10  
CPLX2  
CRHR1  
CSNK1E  
CTDSP1  
CXCL2  
CYBA  
CYP2C9  
DDX42  
DLGAP2  
DNAH9  
DNM2  
DPYSL5  
DUSP3  
DYNC1I1  
E2F7  
EGR3  
EIF4EBP2  
ELAVL2  
EMC3  
ENPEP  
EPHB1  
ERBB4  
ESRRB  
EXOC7  
F11R  
FANCD2  
FBXL17  
FGFBP1  
FMO3  
FOXE3  
FRS2  
FXYD3  
GATA3  
GDF10  
GFRA1  
GGT1  
GLIS1  
GNB5  
GPAT3  
GRIP2  
GSTT1  
GUSB  
HAMP  
HCAR2  
HEY2  
HIST1H2BJ  
HNRNPK  
HTR6  
IDO1  
IFITM5  
IGFBP7  
IKZF2  
IL11RA  
IRAK3  
ISL1  
ITGAE  
JPH3  
KAZN  
KCNK10  
KITLG  
KLHL2  
LCP1  
LHCGR  
LRRC8A  
</textarea>
<p><a href="https://maayanlab.cloud/Enrichr/#" target="_blank">Visit Enrichr</a></p>
<aside class="notes">
<p>Problems here: - Background is offered as a ‚Äúnew option‚Äù not a requirement WHICH IT IS - The website doesn‚Äôt state what statistical test is being done. It refers to this set of tools as ‚Äúgene set enrichment analysis‚Äù which is confusing, because GSEA is a specific statistical algorithm invented for pathway analysis, and this isn‚Äôt it - Databases are out of date, and no easy way to record what was used (not reproducible) -No guide to interpretation ‚Äì results are ranked by adjusted p-value. Let‚Äôs talk about p-value adjustment. When you run multiple tests, you should adjust your p-values because the more tests you perform, the more likely you are to get significant results by chance alone. But even with adjustment, consider this: there are 625 pathways in this database. With an adjusted p-value threshold of 0.05, we‚Äôd expect around 31 false positives, just by random chance.</p>
<p>In this case, we happen to know that our list of 100 genes is, as best we can estimate, unrelated ‚Äî so we‚Äôre not surprised that some results still come up significant. But what if you do believe your genes are biologically related? What if you‚Äôre excited by these results and decide to include them in a paper or presentation?</p>
<p>There‚Äôs no rule saying you can‚Äôt do that ‚Äî but this example shows why you might want to pause and reflect. Tools like this are incredibly useful, but interpreting them wisely takes context and caution. We‚Äôll walk through how to use these tools to generate insight, not just output.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="figure-three-methods-overview">Figure: Three Methods Overview</h3>

<img data-src="images/figure1.jpg" class="r-stretch"><aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="over-representation-analysis-ora">Over-representation analysis (ORA):</h3>
<p>Examine whether any pathways are observed in a gene list of interest more than expected by chance compared with a background set</p>
<ul>
<li>Uses DEGs and a background</li>
<li>Compares observed vs expected overlap</li>
<li>P-values are adjusted for multiple-testing!</li>
</ul>

<img src="images/figure1A.png" width="30%" class="r-stretch"><aside class="notes">
<p>Kalyanee. ORA methods use a hypergeometric test (or very closely-related Fisher‚Äôs test) to test whether the overlap between two lists are more than is expected by change. Over-representation-based methods are conceptually straightforward but have several limitations, such as assuming independence of each gene and requiring an arbitrary cutoff to define differentially expressed gene sets.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="gene-set-enrichment-analysis-gsea">Gene Set Enrichment Analysis (GSEA):</h3>
<p>First rank the total gene set on the basis of detected signals, such as change of gene expression, then tests whether genes annotated to the same pathway tend to cluster together at the top (or bottom) of the ranked list.</p>
<ul>
<li>Uses a full ranked list of genes</li>
<li>Captures subtle, coordinated effects</li>
<li>P-values are adjusted for multiple-testing!</li>
</ul>

<img src="images/figure1B.png" width="30%" class="r-stretch"><aside class="notes">
<p>Kalyanee. Give a brief overview</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="topology-based-analysis">Topology-based analysis:</h3>
<div style="float: right; margin-left: 15px; margin-bottom: 15px;">
<p><img src="images/figure1C.png" width="300px"></p>
</div>
<p>Account for additional information that impacts pathway activity by integrating scores measuring gene positions within a pathway and gene‚Äìgene interactions into the enrichment tests.</p>
<ul>
<li><p>Aim to increase the sensitivity of pathway enrichment analysis by considering genes‚Äô ‚Äúco-expression‚Äù</p></li>
<li><p>Requires experimental evidence for pathway structures and gene‚Äìgene interactions</p></li>
</ul>
<div style="clear: both;">

</div>
<aside class="notes">
<p>Kalyanee. Give a brief overview</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="general-workflow">General Workflow</h3>
<ul>
<li>Start with omics data (e.g., RNA-seq, proteomics, etc.)</li>
<li>Choose appropriate annotation database (e.g., GO, KEGG)</li>
<li>Select statistical method based on data type</li>
<li>Perform PEA and visualize</li>
<li>Document all assumptions and parameters</li>
</ul>

<img data-src="images/figure2.jpg" class="r-stretch"><aside class="notes">
<p>Kalyanee</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="classic-input-transcriptomics">Classic Input: Transcriptomics</h3>
<ul>
<li>DEGs from bulk RNA-seq or microarrays</li>
<li>Common methods: ORA (Fisher), GSEA, TPEA</li>
<li>Caution: arbitrary DEG thresholds may distort findings</li>
<li>GSEA uses full data, reducing cutoff bias</li>
</ul>
<aside class="notes">
<p>Kalyanee</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="the-importance-of-input-and-background-sets">The Importance of Input and Background Sets</h3>
<ul>
<li><strong>Input set:</strong> filtered list of genes, proteins, or metabolites</li>
<li><strong>Background set:</strong> all features detected in the experiment</li>
<li>‚ùó Using all genes in the genome as background = common mistake</li>
</ul>
<aside class="notes">
<p>Kalyanee</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="example-why-background-matters">Example: Why Background Matters</h3>
<ul>
<li>RNA-seq study using genome-wide vs.&nbsp;expressed-gene background</li>
<li>Only 44% overlap in enriched pathways</li>
<li>Use actual detected features as background!</li>
</ul>

<img src="images/F1.large.jpg" width="39%" class="r-stretch"><aside class="notes">
<p>Kalyanee</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="reference-annotation-databases">Reference Annotation Databases</h3>
<ul>
<li>KEGG, Reactome, GO, MetaCyc, EcoCyc, etc.</li>
<li>Pathway size and definitions vary between databases</li>
<li>Use the most updated versions</li>
<li>Report: database, version, date</li>
</ul>
<aside class="notes">
<p>Kalyanee</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<p><strong>Web Tool Summary Table</strong></p>
<div style="max-height: 600px; overflow-y: auto;">
<div style="font-size: 65%; line-height: 1.1;">
<table class="caption-top" style="width:100%;">
<colgroup>
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
<col style="width: 11%">
</colgroup>
<thead>
<tr class="header">
<th>Tool</th>
<th style="text-align: center;">ORA</th>
<th style="text-align: center;">GSEA</th>
<th style="text-align: center;">Topology</th>
<th style="text-align: center;">GO</th>
<th style="text-align: center;">KEGG</th>
<th style="text-align: center;">Reactome</th>
<th style="text-align: center;">MSigDB</th>
<th>Other Databases</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>g:Profiler</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">üî∂</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td>TRANSFAC, miRTarBase, WikiPathways</td>
</tr>
<tr class="even">
<td>Enrichr</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td>ChEA, DrugMatrix, TF/miRNA</td>
</tr>
<tr class="odd">
<td>DAVID</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td>Panther, BioCarta</td>
</tr>
<tr class="even">
<td>WebGestalt</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td>WikiPathways, user-defined sets</td>
</tr>
<tr class="odd">
<td>Reactome</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td></td>
</tr>
<tr class="even">
<td>PantherDB</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td>Panther Pathways</td>
</tr>
<tr class="odd">
<td>Metascape</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td>CORUM, WikiPathways</td>
</tr>
<tr class="even">
<td>ShinyGO</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">üî∏</td>
<td>Limited subset of MSigDB</td>
</tr>
<tr class="odd">
<td>PathDIP</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td>PID, BioCarta, PPI-aware pathways</td>
</tr>
<tr class="even">
<td>GSEA-MSigDB</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td>Hallmark, C1‚ÄìC7 collections</td>
</tr>
<tr class="odd">
<td>ExpressAnalyst</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;">‚úÖ</td>
<td>BioCarta, WikiPathways</td>
</tr>
<tr class="even">
<td>Cytoscape EnrichMap</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">‚úÖ</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">Any</td>
<td style="text-align: center;">Any</td>
<td style="text-align: center;">Any</td>
<td style="text-align: center;">Any</td>
<td>Visualization</td>
</tr>
</tbody>
</table>
</div>
</div>
<aside class="notes">
<p>Kalyanee</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="quick-tips-to-avoid-pea-pitfalls">9 Quick Tips to Avoid PEA Pitfalls</h3>
<ol type="1">
<li>Know what analysis type fits your data</li>
<li>Clean, validated input gene list only</li>
<li>Use &gt;1 PEA tool, compare results</li>
<li>Document all tool versions &amp; parameters</li>
<li>Always use adjusted p-values</li>
<li>Choose statistical tests &amp; visualizations wisely</li>
<li>Consider analyzing gene subgroups/networks</li>
<li>Validate with recent literature</li>
<li>Review with a wet-lab biologist or clinician</li>
</ol>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="common-errors-in-published-pea-studies">Common Errors in Published PEA Studies</h3>
<ul>
<li>90% of surveyed studies used incorrect background sets</li>
<li>16 of 25 popular tools used outdated databases</li>
<li>Up to 40% more false positives without multiple testing correction</li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="visualizing-your-results">Visualizing Your Results</h3>
<ul>
<li>Visualization improves interpretation</li>
<li>Tools:
<ul>
<li>enrichplot (R)</li>
<li>Enrichment Map (Cytoscape)</li>
<li>REVIGO</li>
<li>CirGO</li>
</ul></li>
<li>Use clustering or similarity to group redundant terms</li>
</ul>
<aside class="notes">
<p>Heather<br>
Clustering terms can reduce redundancy<br>
REVIGO is great for summarizing GO terms.<br>
Enrichment Map is ideal for interactive exploration of results.</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="reporting-is-key">Reporting is Key</h3>
<ul>
<li>The result of pathway enrichment depends on the data, assumptions, and tools you use.</li>
<li>Therefore, these <strong>must</strong> be reported in your presentations and publications:
<ul>
<li>Gene list origin and filtering criteria</li>
<li>Background set definition</li>
<li>Tool name, version, database, and date</li>
<li>Statistical method, FDR threshold</li>
</ul></li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="special-case-scrna-seq">Special Case: scRNA-seq</h3>
<ul>
<li>Sparse data, many zeros</li>
<li>Tools like iDEA use Bayesian models to improve detection</li>
<li>Enables pathway analysis at single-cell resolution</li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="special-case-epigenomics">Special Case: Epigenomics</h3>
<ul>
<li>Gene length and CpG mapping introduce bias</li>
<li>Tools: ebGSEA, GOmeth</li>
<li>GOmeth accounts for multigene mapping and gene size</li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="integrated-omics">Integrated Omics</h3>
<ul>
<li>Combines transcriptomics, proteomics, GWAS, etc.</li>
<li>Tool: ActivePathways (combines p-values from each omics layer)</li>
<li>Pros: improves power</li>
<li>Cons: complex, needs careful integration</li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="final-thoughts">Final Thoughts</h3>
<ul>
<li>Pathway enrichment is powerful if done right</li>
<li>Many omics types require special considerations</li>
<li>Don‚Äôt rely on defaults or black-box tools</li>
<li>Collaboration between biologists and bioinformaticians is key</li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<h3 id="open-questions">Open Questions</h3>
<ul>
<li>What are the best ways to integrate multiple data layers?</li>
<li>What makes a good PEA tool for non-programmers?</li>
</ul>
<aside class="notes">
<p>Heather</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
</section>
<section class="slide level2">

<p>:::::::::::::</p>

</section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="Presentation.part1and3.Heather_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>